runtime/src/identity.rs

```diff
@@ -105,6 +115,38 @@ pub enum LinkedKeyInfo {
     Group(Vec<IdentityId>),
 }
 
+/// It represents an authorization that any account could sing to allow operations related with a
+/// target identity.
+///
+/// # Safety
+///
+/// Please note, that `nonce` has been added to avoid **reply attack** and it should be the current
+/// value of nonce of master key of `target_id`. See `System::account_nonce`.
+/// In this way, the authorization is delimited to an specific transaction (usually the next one)
+/// of master key of target identity.
+#[derive(codec::Encode, codec::Decode, Clone, PartialEq, Eq, Debug)]
+pub struct TargetIdAuthorization<N> {
+    /// Target identity which is authorized to make an operation.
+    pub target_id: IdentityId,
+    /// It HAS TO be the `System::account_nonce()` of the master key of `target_id`.
```

Comment:
I do not think that this should be the case. `target_id` should be allowed to freely do other transactions after asking a signing key to sign data and before processing that signed data. i.e. Data being signed should not use the target's nonce.

Perhaps we should add a mapping (signer, nonce) -> bool [nonce used or not] in the storage.

Reply:
From slack chat:
- Use an Identity off-chain authorization nonce. This nonce is autoincremented in each authorization. It allows the identity to execute other TXs meanwhile it is waiting for signers.
- Authorization has an `expires_at` field, to limit that off-chain authorization.
- Signer can revoke any authorization immediately: `Identity::revoke_offchain_authorization`.